k8s-daemonset/k8s/servicemesh.yml

```diff
@@ -0,0 +1,356 @@
+      - kind: io.l5d.k8s.daemonset
+        namespace: linkerd
+        port: http-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+    - kind: io.l5d.k8s
+      prefix: /io.l5d.k8s.h2
+      transformers:
+        # The daemonset transformer replaces the address of the target app with
+        # the address of the h2-incoming router of the Linkerd daemonset pod
+        # on the target app's node.
+      - kind: io.l5d.k8s.daemonset
+        namespace: linkerd
+        port: h2-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+    - kind: io.l5d.k8s
+      prefix: /io.l5d.k8s.grpc
+      transformers:
+        # The daemonset transformer replaces the address of the target app with
+        # the address of the grpc-incoming router of the Linkerd daemonset pod
+        # on the target app's node.
+      - kind: io.l5d.k8s.daemonset
+        namespace: linkerd
+        port: grpc-incoming
+        service: l5d
+        # hostNetwork: true # Uncomment if using host networking (eg for CNI)
+    - kind: io.l5d.rewrite
+      prefix: /portNsSvcToK8s
+      pattern: "/{port}/{ns}/{svc}"
+      name: "/k8s/{ns}/{port}/{svc}"
+
+    # Telemeters export metrics and tracing data about Linkerd, the services it
+    # connects to, and the requests it processes.
+    telemetry:
+    - kind: io.l5d.prometheus # Expose Prometheus style metrics on :9990/admin/metrics/prometheus
+    - kind: io.l5d.recentRequests
+      sampleRate: 0.25 # Tune this sample rate before going to production
+    # - kind: io.l5d.zipkin # Uncomment to enable exporting of zipkin traces
+    #   host: zipkin-collector.default.svc.cluster.local # Zipkin collector address
+    #   port: 9410
+    #   sampleRate: 1.0 # Set to a lower sample rate depending on your traffic volume
+
+    # Usage is used for anonymized usage reporting.  You can set the orgId to
+    # identify your organization or set `enabled: false` to disable entirely.
+    usage:
+      orgId: linkerd-examples-servicemesh
+
+    # Routers define how Linkerd actually handles traffic.  Each router listens
+    # for requests, applies routing rules to those requests, and proxies them
+    # to the appropriate destinations.  Each router is protocol specific.
+    # For each protocol (HTTP, HTTP/2, gRPC) we define an outgoing router and
+    # an incoming router.  The application is expected to send traffic to the
+    # outgoing router which proxies it to the incoming router of the Linkerd
+    # running on the target service's node.  The incoming router then proxies
+    # the request to the target application itself.  We also define HTTP and
+    # HTTP/2 ingress routers which act as Ingress Controllers and route based
+    # on the Ingress resource.
+    routers:
+    - label: http-outgoing
+      protocol: http
+      servers:
+      - port: 4140
+        ip: 0.0.0.0
+      dtab: |
```

Comment:
Maybe a comment like this?

This dtab looks up service names in k8s and falls back to DNS if they're not found (e.g. for external services). It accepts names of the form "service" and "service.namespace", defaulting the namespace to "default". For DNS lookups, it uses port 80 if unspecified. Note that dtab rules are read bottom to top. To see this in action, on the Linkerd administrative dashboard, click on the "dtab" tab, select "http-output" from the dropdown, and enter a service name like "a.b". (Or click on the "requests" tab to see recent traffic through the system and how it was resolved.)